src/Microsoft.AspNetCore.Http/Features/ResponseCookiesFeature.cs

```diff
@@ -25,8 +49,17 @@ public IResponseCookies Cookies
                 if (_cookiesCollection == null)
                 {
                     var headers = HttpResponseFeature.Headers;
-                    _cookiesCollection = new ResponseCookies(headers);
+
+                    var serviceProvider = ServiceProvidersFeature.RequestServices;
+                    var provider = (ObjectPoolProvider)serviceProvider.GetService(typeof(ObjectPoolProvider)) ??
+                        new DefaultObjectPoolProvider();
+                    var pool = provider.CreateStringBuilderPool(
```

Comment:
Does this actually create a new pool per request? Or is it a GetOrCreate?  How about if you call DefaultObjectPoolProvider?

If so, how is this an improvement over creating a few StringBuilders per request?